ckanext/external_storage/tests/test_actions.py

```diff
@@ -0,0 +1,72 @@
+from ckan.tests import factories
+from ckan.tests import helpers as test_helpers
+
+from ckanext.external_storage.tests import get_context
+
+
+class TestBlobSotrageActions():
+    """Test cases for logic actions
+    """
+
+    org_admin = factories.User()
+    initial_dataset = factories.Dataset()
+
+    first_resource = factories.Resource(
+        name="First Resource",
+        package_id=initial_dataset['id'],
+        sample="{u'1': {u'VMP_SNOMED_CODE': u'4747111000001104', u'ODS_CODE': u'REF'}}",
+        schema="{u'fields': [{u'type': u'string'}], u'missingValues': [u'']}"
+    )
+
+    second_resource = factories.Resource(
+        name="First Resource",
+        package_id=initial_dataset['id'],
+        sample={'1': {'VMP_SNOMED_CODE': '4747111000001104'}},
+        schema={'fields': [{'type': 'string'}], 'missingValues': ['']}
+    )
+
+    def test_resource_schema_show(self):
+        """Test if the resource.schema is a string the resource_sample_show should return JSON object
+        """
+        context = get_context(self.org_admin)
```

Comment:
You don't need `get_context` (see above) here. It would be cleaner to use the existing `user_context(user)`  context manager:

```python
def test_resource_schema_show(self):
    with user_context(self.org_admin) as context:
        resource_schema = test_helpers.call_action(
            'resource_schema_show',
            context,
            id=self.first_resource['id'])
    expect = {'fields': [{'type': 'string'}], 'missingValues': ['']}
    assert resource_schema == expect
```

BTW, context managers are an awesome Python feature. You can read about them more for example here: https://towardsdatascience.com/the-magic-of-python-context-managers-adb92ace1dd0. Understanding how they work and when to use them is a very useful tool for Python developers.